src/serializer/serializer.js

```diff
@@ -1629,24 +1589,27 @@ export class Serializer {
 
       let location;
       for (let value of initializedValues) {
-        location = this.refs.get(value);
-        if (location !== undefined) break;
+        if (!value.mightBeUndefined()) {
+          location = this.refs.get(value);
+          if (location !== undefined) break;
+        }
       }
       if (location === undefined) {
         // Take care of side-effecting code that is associated with an inlined value
-        location = this.allocateIndexedVar();
+        location = t.identifier(this.valueNameGenerator.generate("initialized"));
+        this.mainBody.push(t.variableDeclaration("var", [t.variableDeclarator(location)]));
         initializationStatements.unshift(
           t.expressionStatement(
             t.assignmentExpression(
               "=",
               location,
-              Serializer.voidExpression
+              t.nullLiteral()
```

Comment:
I'll add the following comments:

We have some initialization code, and it should only get executed once, so we are going to guard it.
First, let's see if one of the initialized values is guaranteed to not be undefined after initialization. In that case, we can use that state-change to figure out if initialization needs to run. 
Second, if we didn't find a non-undefined value, let's make one up. It will transition from `undefined` to `null`.